docs/swagger/post.yaml

```diff
@@ -68,7 +68,7 @@ paths:
               - data
             properties:
               data:
-                $ref: "#/definitions/Post"
+                $ref: "#/definitions/PostForm"
```

Comment:
I don't like the need to have two definitions. Post and PostForm which are almost identical.

This is just only due to `Content` and `ContentForm`?

Couldn`t we integrate the Topics with in `ContentMetadata`?

Reply:
@luke- We cannot use the same object/structure for a request data and for a response data.
For example, response data contains `id`, but we should not allow to send/request this field because it cannot be updated. Also we should not allow to update other fields like `guid`, `created_at` which exist in the object `ContentMetadata`.
So I decided to create the separate object `ContentForm` in order to keep there only fields which are allowed in request data.

Reply:
@yurabakhtin It is possible to flag attributes in Swagger with `readOnly` or `writeOnly`.  
https://swagger.io/docs/specification/data-models/data-types/

e.g. for `id` we already use this:
https://github.com/humhub/rest/blob/master/docs/swagger/content.yaml#L212

Reply:
@luke- I tested the flags `readOnly` or `writeOnly`, but a generated HTML file doesn't hide field depending on these flags in request/response:

![test_read_only](https://user-images.githubusercontent.com/6995524/166492544-e74d87c6-468f-4c80-beae-81eef29e1c30.png)

But we need this:

![array_types](https://user-images.githubusercontent.com/6995524/166492603-3e329bd7-5e6f-46bd-845c-1a9c5706051e.png)

I.e. in request and response we have the field with key `topics` but it has a different format.

Reply:
@luke- Please review my changes in the commit https://github.com/humhub/rest/pull/86/commits/4f3d9d9b347df75b477eafd8c810416ea7e557cc

I found how to use `readOnly` and `writeOnly` flags only for definitions/components but I cannot find how to use these flags for different examples but with same key, as we need for property/key `topics`.

For request part we need only array on strings:

- `["First topic", "Second tag"]`

For response - array of objects:

```
[
  { "id": 1, "name": "First topic" },
  { "id": 2, "name": "Second tag" }
]
```

So in the commit I created a new key `topics-array`, but in php code we still have `topics` for request data.
If we stop on this solution we should modify php code and accept `topics-array` on request side.
I.e. as I wrote above the main reason why I created a separate object/component/definition is that `topics` has a different schema in request and response parts.

Reply:
@yurabakhtin Thanks the commit looks fine. But can't we just use the "Id" in the Topic example as ReadOnly. Or did is missed something here?

For reading:

```
[
  { "id": 1, "name": "First topic" },
  { "id": 2, "name": "Second tag" } }
]
```
Writing:
[
  { "name": "First topic" },
  { "name": "Second tag" }
]

I would also like it if the `ContentMetadata` were not completely `readOnly`, but you could control the Visibility & Archived & Pinned there.